### DISCUSSION TITLE
Wanted - Hired gun

### DISCUSSION THREAD

**Discussion by chazhenry** [airsdk/Adobe-Runtime-Support#1674] [Q&A]
I have a working app built from 743 and compiling to Mac and PC. I use IntelliJ and understand little of the compile pipeline. I've hacked in xml tweaks, mac entitlements and more. Somehow it works. 

Trying to port it to Android and _nothing_ works. Tried to compile the HelloWorld sample - still no. I get an APK, but think its really an AAB. Too many issues to list, but the bottom line is that I don't understand enough of the process to fix it. 

Looking for someone to help and willing to pay. Many questions. Should I be compiling with the command line? What IDE should I be using? Android Studio work? Do I need android specific XML entries? How to test android if I can't create APK (can't sideload the AAB)? If you're able and willing, let me know.

**Comment by marchbold** [comment]
From intellij currently it will be an apk unless you change the override setting in the AIR SDK `lib/adt.cfg`. i.e. if you have uncommented the `CreateAndroidAppBundle=true` line:

```
# To override the creation of an Android APK file, and to instead create an Android App Bundle, the below
# flag can be uncommented. Note that the bundle may end up with an "apk" file extension
#CreateAndroidAppBundle=true
``` 

Unless you have done that, then you will be getting an apk. 

What exactly is failing, are you getting an app installed on a device? or is the build failing?

**Comment by chazhenry** [comment]
Giving up on Android. Can't sideload a SWF with its code intact. Works on PC/Mac. Moving on. 

Now need help codesigning the Mac app. Will use cert from Apple developer account. Will pay for help. 

I'm properly injecting entitlements into the app bundle and building a PKG installer, but Mac warns and you must relax via:
`Apple > System Preferences > Security > General tab > Click “Open Anyway”`

Would love to get those chains off of the app.

**Reply by ajwfrost** [reply]
See #1688 and its links for some info. Essentially you will need a "Developer ID Application" certificate to code-sign it, and the AIR developer tool should help with that. Then you can create a pkg file via pkgbuild using a "Developer ID Installer" certificate, and then you'll need to notarize that.

After all that, your uses will still see a pop-up, but it's along the lines of "This app was downloaded from the internet, are you sure you want to proceed?" and in smaller font "Apple has checked this software for malicious code" or similar..

thanks

**Reply by chazhenry** [reply]
@ajwfrost Using IntelliJ, I compile my project to a captive runtime bundle. That gives me a .app file that has the typical app structure - Contents > Resources, info.plist, etc. I currently run a bash script that issues 'codesign' and gives me camera and microphone entitlements. 

I now have an entitled, runnable app. I assume that I need to codesign that. 

Then, I use Packager to get a PKG. I understand that I need to codesign, register and staple that.  I've read that the latest SDKs have eliminated the need to codesign some of the Content of the .app. Is #1688 still relevant?  

I have made certs from my Apple dev account. Would still like to find someone who could bash script this so I could run it after each new compile. I think I have all the pieces, but don't know that keychain and other Mac specific pieces I've encountered in the various notes. 

Any takers?
